odoo/tools/config.py

```diff
@@ -559,6 +567,25 @@ def die(cond, msg):
         ]
         return opt
 
+    def _parse_size(self, text):
+        match = re.match(r'^\s*(?P<number>\d+(\.\d+)?)\s*(?P<unit>(?P<exponent>\w)?(?P<base>i)?B?)\s*$', text)
```

Comment:
> This pattern allows `1.2465` even though it makes absolutely no sense.
> 
> Also why does `unit` get a named non-optional capturing group? It should be a non-capturing optional group instead, and the unit should be non-optional.
> 
> This would very much benefit from `re.VERBOSE` and comments.

Fair. Let me take this one back to my bench and clean it up.

Reply:
I ripped out support for decimal prefixes and floats and the regular expression is now verbose and much more understandable.